Automattic/jetpack
# Infinite scroll added checks to validate input

Fixes Warnings in Infinite scroll Module like: 
`Undefined array key "query_before"`
`Undefined array key "query_args"`
`Undefined array key "last_post_date"`
118d8bdb2d4600db50a5c7b306828911-logstash
## Proposed changes:

* Added isset check to validate inputs. Removed phpcs ignores, accordingly.
* Deprecate the `infinite_scroll_posts_where` hook

## Testing instructions:






Code Review

Edited: For the changes when deprecating the `infinite_scroll_posts_where`. 
- Please make sure to test [Infinite Scroll](https://jetpack.com/support/infinite-scroll/) with a non-block theme and activate Infinite Scroll module.
- Make sure you have enough posts on the test site. Go to the frontend and scroll to the bottom of the posts so that the code in this PR gets triggered.
- Ensure we don't get any errors
- Test with this onebin 35e05-pb/ to check that the filter still works and you should see a Depecrated hook like the following
```
PHP Deprecated:  Hook infinite_scroll_posts_where is <strong>deprecated</strong> since version $$next-version$$ with no alternative available. in /var/www/html/wp-includes/functions.php on line 6085
```
